ddl/placement/rule.go

```diff
@@ -84,18 +84,17 @@ func NewRules(role PeerRoleType, replicas uint64, cnstr string) ([]*Rule, error)
 
 	constraints1, err1 := NewConstraintsFromYaml(cnstbytes)
 	if err1 == nil {
-		// can not emit REPLICAS with an array or empty label
-		if replicas == 0 {
-			return rules, fmt.Errorf("%w: should be positive", ErrInvalidConstraintsRelicas)
-		}
-
 		rules = append(rules, NewRule(role, replicas, constraints1))
 		return rules, nil
 	}
 
 	constraints2 := map[string]int{}
 	err2 := yaml.UnmarshalStrict(cnstbytes, &constraints2)
 	if err2 == nil {
+		if replicas != 0 {
+			return rules, fmt.Errorf("%w: should not specify replicas=%d when using dict syntax", ErrInvalidConstraintsRelicas, replicas)
+		}
+
 		ruleCnt := 0
```

Comment:
ruleCnt seems useless now, can we remove it?

Reply:
Rest LGTM

Reply:
Removed